pyracf/common/utilities/logger.py

```diff
@@ -203,27 +221,38 @@ def redact_result_xml(
     ) -> str:
         """
         Redacts a list of specific secret traits in a result xml string.
-        Based on the following RACF command pattern:
+        Based on the following RACF command patterns:
             'TRAIT (value)'
+            "TRAIT ('value')"
         This function also accounts for varied amounts of whitespace in the pattern.
         """
         if isinstance(security_result, list):
             return security_result
         for xml_key in secret_traits.values():
             racf_key = xml_key.split(":")[1] if ":" in xml_key else xml_key
-            end_pattern = ")"
             if isinstance(security_result, bytes):
                 match = re.search(
                     rf"{racf_key.upper()} +\(", security_result.decode("cp1047")
                 )
-                end_pattern = end_pattern.encode("cp1047")
             else:
                 match = re.search(rf"{racf_key.upper()} +\(", security_result)
             if not match:
                 continue
-            security_result = self.__redact_string(
-                security_result, match.end(), end_pattern
-            )
+            security_result = self.__redact_string(security_result, racf_key)
+            if isinstance(security_result, bytes):
+                security_result = re.sub(
+                    rf"<message>([A-Z]*[0-9]*[A-Z]) [^<>]*{racf_key.upper()}[^<>]*<\/message>",
+                    rf"<message>REDACTED MESSAGE CONCERNING {racf_key.upper()}, "
+                    + r"REVIEW DOCUMENTATION OF \1 FOR MORE INFORMATION</message>",
+                    security_result.decode("cp1047"),
+                ).encode("cp1047")
+            else:
+                security_result = re.sub(
+                    rf"<message>([A-Z]*[0-9]*[A-Z]) [^<>]*{racf_key.upper()}[^<>]*<\/message>",
+                    rf"<message>REDACTED MESSAGE CONCERNING {racf_key.upper()}, "
+                    + r"REVIEW DOCUMENTATION OF \1 FOR MORE INFORMATION</message>",
+                    security_result,
+                )
```

Comment:
Also explain how these regexes work.

Reply:
Added the following:

        This function employs the following regular expression:
        <message>([A-Z]*[0-9]*[A-Z]) [^<>]*{racf_key.upper()}[^<>]*<\/message>" - 
        Designed to match the above pattern by starting and ending with the message xml tag 
        string as shown, the value of the message is targeted based on the racf_key. This
        should capture only messages that contain information about a redacted key.
        The message identifier is "captured" as a variable and preserved by the substitution
        operation through the use of the \1 supplied in the replacement string.

Reply:
Just a slight working change suggestion:

```
Designed to match any message xml element that contains the provided racf_key.
```

Instead of
```
Designed to match the above pattern by starting and ending with the message xml tag string as shown, the value of the message is targeted based on the racf_key.
```